src/ert/_c_wrappers/enkf/observations/obs_vector.py

```diff
@@ -16,89 +13,58 @@
     from ert._c_wrappers.enkf.enkf_obs import EnkfObs
 
 
-class ObsVector(BaseCClass):
-    TYPE_NAME = "obs_vector"
-
-    _alloc = ResPrototype(
-        "void* obs_vector_alloc(enkf_obs_impl_type, char*, char*, int)",
-        bind=False,
-    )
-    _free = ResPrototype("void  obs_vector_free( obs_vector )")
-    _get_state_kw = ResPrototype("char* obs_vector_get_state_kw( obs_vector )")
-    _get_key = ResPrototype("char* obs_vector_get_key( obs_vector )")
-    _iget_node = ResPrototype("void* obs_vector_iget_node( obs_vector, int)")
-    _get_num_active = ResPrototype("int   obs_vector_get_num_active( obs_vector )")
-    _iget_active = ResPrototype("bool  obs_vector_iget_active( obs_vector, int)")
-    _get_impl_type = ResPrototype(
-        "enkf_obs_impl_type obs_vector_get_impl_type( obs_vector)"
-    )
-    _install_node = ResPrototype(
-        "void  obs_vector_install_node(obs_vector, int, void*)"
-    )
-    _get_next_active_step = ResPrototype(
-        "int   obs_vector_get_next_active_step(obs_vector, int)"
-    )
-    _get_obs_key = ResPrototype("char*  obs_vector_get_obs_key(obs_vector)")
-
+class ObsVector:
     def __init__(
         self,
         observation_type: EnkfObservationImplementationType,
         observation_key: str,
         config_node_key: str,
-        num_reports: int,
+        num_steps: int,
     ):
-        assert isinstance(observation_type, EnkfObservationImplementationType)
-        assert isinstance(observation_key, str)
-        assert isinstance(config_node_key, str)
-        assert isinstance(num_reports, int)
-        c_ptr = self._alloc(
-            observation_type, observation_key, config_node_key, num_reports
-        )
-        super().__init__(c_ptr)
+        self.observation_type = observation_type
+        self.observation_key = observation_key
+        self.config_node_key = config_node_key
+        self.nodes: List[Optional[Union[SummaryObservation, GenObservation]]] = [
+            None
+        ] * num_steps
+        self.steps = set()
 
     def getDataKey(self) -> str:
-        return self._get_state_kw()
+        return self.config_node_key
 
     def getObservationKey(self) -> str:
-        return self.getKey()
+        return self.observation_key
 
     def getKey(self) -> str:
-        return self._get_key()
+        return self.observation_key
 
     def getObsKey(self) -> str:
-        return self._get_obs_key()
+        return self.observation_key
```

Comment:
`getObservationKey`, `getKey` and `getObsKey` all return the same string.

This __repr__ does not make much sense:
```
f"key = {self.getKey()}, obs_key = {self.getObsKey()}, "
```

Reply:
Changed the repr to be the same as the constructor. These are weird, but did actually return the same thing. I was thinking to clean up the usage of these in a separate step.